pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -21,50 +36,491 @@ func init() {
+	if _, err := builder.WriteString(fmt.Sprintf("package policy\nallow_all := %v\n", p.AllowAll)); err != nil {
+		return "", err
+	}
+
+	if err := addContainers(builder, p.Containers); err != nil {
+		return "", err
+	}
+
+	return builder.String(), nil
+}
+
+func createRegoEnforcer(state SecurityPolicyState, defaultMounts []oci.Mount, privilegedMounts []oci.Mount) (SecurityPolicyEnforcer, error) {
+	policy, err := state.SecurityPolicy.toInternal()
+	if err != nil {
+		return nil, fmt.Errorf("error converting to internal format: %w", err)
+	}
+
+	regoPolicy, err := newRegoPolicyFromInternal(policy, defaultMounts, privilegedMounts)
+	if err != nil {
+		return nil, fmt.Errorf("error converting to Rego: %w", err)
+	}
+
+	regoPolicy.base64policy = state.EncodedSecurityPolicy.SecurityPolicy
+	return regoPolicy, nil
+}
+
+func newRegoPolicyFromInternal(securityPolicy *securityPolicyInternal, defaultMounts []oci.Mount, privilegedMounts []oci.Mount) (*RegoEnforcer, error) {
+	policy := new(RegoEnforcer)
+	policy.behavior = PolicyCode
+	if code, err := securityPolicy.MarshalRego(); err == nil {
+		policy.objects = code
+	} else {
+		return nil, fmt.Errorf("failed to convert json to rego: %w", err)
+	}
+
+	defaultMountData := make([]interface{}, 0, len(defaultMounts))
+	privilegedMountData := make([]interface{}, 0, len(privilegedMounts))
+	policy.data = map[string]interface{}{
+		"started":          []string{},
+		"devices":          map[string]string{},
+		"containers":       map[string]interface{}{},
+		"defaultMounts":    appendMountData(defaultMountData, defaultMounts),
+		"privilegedMounts": appendMountData(privilegedMountData, privilegedMounts),
+		"sandboxPrefix":    guestpath.SandboxMountPrefix,
+		"hugePagesPrefix":  guestpath.HugePagesMountPrefix,
+	}
+	policy.mutex = new(sync.Mutex)
+	policy.base64policy = ""
+
+	modules := map[string]string{
+		"behavior.rego":  policy.behavior,
+		"objects.rego":   policy.objects,
+		"framework.rego": FrameworkCode,
+	}
+
+	// TODO temporary hack for debugging policies until GCS logging design
+	// and implementation is finalized. This option should be changed to
+	// "true" if debugging is desired.
+	options := ast.CompileOpts{
+		EnablePrintStatements: false,
+	}
+
+	if compiled, err := ast.CompileModulesWithOpt(modules, options); err == nil {
+		policy.compiledModules = compiled
+	} else {
+		return nil, fmt.Errorf("rego compilation failed: %w", err)
+	}
+
+	return policy, nil
+}
+
+func (policy RegoEnforcer) Query(input map[string]interface{}) (rego.ResultSet, error) {
+	store := inmem.NewFromObject(policy.data)
+
+	var buf bytes.Buffer
+	rule := input["name"].(string)
+	query := rego.New(
+		rego.Query(fmt.Sprintf("data.policy.%s", rule)),
+		rego.Compiler(policy.compiledModules),
+		rego.Input(input),
+		rego.Store(store),
+		rego.PrintHook(topdown.NewPrintHook(&buf)))
+
+	ctx := context.Background()
+	results, err := query.Eval(ctx)
+	if err != nil {
+		log.G(ctx).WithError(err).WithFields(logrus.Fields{
```

Comment:
It doesn't look like this line is actually logging anything, as it doesn't have a call to `Error`, `Info`, etc.

Reply:
@matajoh is this supposed to be `.Error()` ?

Reply:
Also for nit PR: `objects` is the user's input policy, right? So I think we shouldn't bother logging it here at all (maybe just log it once as `Debug` when we read it in).

Reply:
I made this change based on a recommendation from @anmaxvl. Whatever is the desired convention for this repo should be used.

Reply:
I am fairly certain as written this line does not log anything. It creates a temporary logger object that has an error and fields associated with it, but then we don't call anything like `Error` on it, so that logger just gets thrown away.

Personally I'd be more in favor of just removing this line and logging the generated Rego when we construct the enforcer (as I commented above).